test/nri.bats

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bats
+
+#
+# This file is used to hook each test from that binary into
+# BATS as a separate test case for more granular test result
+# reporting. It needs to be updated whenever test cases are
+# added to or removed from the test binary.
```

Comment:
Makes sense to be more explicit that we mean `NRITEST_BINARY`.

Ideally, there has to be some assertion to make sure that the list of tests is consistent between this file and nri.test.

Reply:
True, I guess you mean 's/from that binary/from NRITEST_BINARY/'

Yes, about the consistency... initially I was generating the NRI bats test using a template and the list of NRI test cases from the client (-test.list Test). Actually the above comment mistake is the result of my obviously sloppy updating of the comment in what originally was the bats template. Anyway, the dynamic generation was disliked in earlier review comments, so I switched to this static one. What I haven't addressed yet for that review comment is adding a check that all listed tests are accounted for. But I'll do that now...

Reply:
Updated. Added Makefile rule and github action for checking test case consitency in test client vs. NRI bats test cases.